src/js/components/AppBody.vue

```diff
@@ -25,22 +25,146 @@
           <template slot="paneL">
             <split-pane split="vertical">
               <template slot="paneL">
-                <single-image class="pane-content single-image-pane"></single-image>
+                <single-image 
+                  class="pane-content single-image-pane"
+                  :corpus="corpus"></single-image>
               </template>
               <template slot="paneR">
                 <editor class="pane-content"></editor>
               </template>
             </split-pane>
           </template>
           <template slot="paneR">
-            <combination class="combination-pane"></combination>
+            <combination 
+              class="combination-pane"
+              :corpus="corpus"></combination>
           </template>
         </split-pane>
       </div>
     </div>
   </div>
 </template>
 
+<script>
+import HeaderMenu from './HeaderMenu.vue'
+import MainMenu from './menu/MainMenu.vue'
+import SplitPane from 'vue-splitpane'
+import SingleImage from './SingleImage.vue'
+import Editor from './editor/Editor.vue'
+import Combination from './Combination.vue'
+
+import Corpus from '~/models/Corpus.js'
+
+export default {
+  components: {
+    'header-menu': HeaderMenu,
+    'main-menu': MainMenu,
+    'split-pane': SplitPane,
+    'single-image': SingleImage,
+    'editor': Editor,
+    'combination': Combination,
+  },
+  data() {
+    return {
+      keepMenuOpen: false,
+      mouseOver: false,
+      corpus: Corpus,
+      menuLoaded: false,
+    }
+  },
+  computed: {
+    menuOpen() {
+      return this.mouseOver || this.keepMenuOpen
+    },
+  },
+  created() {
+    // Create and populate the corpus model, with existing data
+    // from the router.
+    this.$store.commit('addWorking')
+    this.corpus = new Corpus(this.$store.state.sessionID, this.$store.state.userID)
+    this.corpus.populateCombinations()
+    .then(res => {
+      this.menuLoaded = true
+      this.$store.commit('delWorking')
+      if (this.$route.params.scrollID && this.$route.params.scrollID !== '~') {
+        this.$store.commit('addWorking')
+        this.corpus.populateColumnsOfCombination(
+          this.$route.params.scrollID,
+          this.$route.params.scrollVersionID
+        )
+        this.corpus.populateImageReferencesOfCombination(this.$route.params.scrollVersionID)
```

Comment:
Might want consider another approach rather than nesting promises, you should be able to chain them together like so:

```javascript
this.corpus.stepOne().then(res1 => {
  return this.doStepTwo();
})
.then(res2 => {
   return this.doStep3();
})
.catch(err => {
  // handle err for whole chain.
});

```

Reply:
Cool, that is indeed more readable.  I'll make the update.